watertap/core/plugins/solvers.py

```diff
@@ -0,0 +1,135 @@
+###############################################################################
+# WaterTAP Copyright (c) 2021, The Regents of the University of California,
+# through Lawrence Berkeley National Laboratory, Oak Ridge National
+# Laboratory, National Renewable Energy Laboratory, and National Energy
+# Technology Laboratory (subject to receipt of any required approvals from
+# the U.S. Dept. of Energy). All rights reserved.
+#
+# Please see the files COPYRIGHT.md and LICENSE.md for full copyright and license
+# information, respectively. These files are also available online at the URL
+# "https://github.com/watertap-org/watertap/"
+#
+###############################################################################
+
+import pyomo.environ as pyo
+from pyomo.core.base.block import _BlockData
+from pyomo.core.kernel.block import IBlock
+from pyomo.solvers.plugins.solvers.IPOPT import IPOPT
+
+from idaes.core.util.scaling import (constraint_autoscale_large_jac,
+        get_scaling_factor, set_scaling_factor, unset_scaling_factor)
+from idaes.logger import getLogger
+
+_log = getLogger("watertap.core")
+
+
+@pyo.SolverFactory.register("ipopt-watertap",
+        doc="The Ipopt NLP solver, with user-based variable and automatic Jacobian constraint scaling")
+class IpoptWaterTAP(IPOPT):
+
+    def _presolve(self, *args, **kwds):
+        if len(args) > 1 or len(args) == 0:
+            raise TypeError(f"IpoptWaterTAP.solve takes 1 positional argument but {len(args)} were given")
+        if not isinstance(args[0], (_BlockData, IBlock)):
+            raise TypeError("IpoptWaterTAP.solve takes 1 positional argument: a Pyomo ConcreteModel or Block")
+
+        self._set_user_scaling()
+
+        self._tee = kwds.get("tee", False)
+        if self._tee:
+            print("ipopt-watertap: Ipopt with user variable scaling and IDAES jacobian constraint scaling")
```

Comment:
Probably should ignore tee and use a logger instead of prints throughout.

Reply:
Thanks for having a look at this @eslickj. 

I settled on the current approach so there's a single place to set output from the `solve` method, which I think is easier on the user -- in particular reporting the options set by the user in `_get_option`. That said, I'm open to reporting this at a set log-level (or changing the log-level its reported at based on the setting of `tee`).

Reply:
I think that's mostly fine.  I have one concern about what is probably a rare case, and I don't even know if it's problem or not.  If someone where to redirect the solver output, would the print statements get caught?  I guess it depends on if someone just starts grabbing all of stdout or just from the solver subprocess.  I do have a utility to direct solver output to a logger, but don't remember how it works.  I'll check on it.

Reply:
Okay, it uses ```pyomo.common.tee.capture_output()```, and as far as I can tell it would capture the prints.  So I have no reason to object to the print statements.